extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultProxy.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.vault;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@Path("/v1")
+@RegisterRestClient
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface VaultProxy extends VaultBaseProxy {
```

Comment:
Please see the comments on the VaultBaseProxy. Perhaps having `VaultUserPassProxy` (with the userpass specific login) and `VaultKubernetesProxy` (with the K. specific login operations) would be better ?

Reply:
see https://github.com/quarkusio/quarkus/pull/2897#discussion_r312193405